libsupport/include/katana/Random.h

```diff
@@ -8,15 +8,15 @@
 
 namespace katana {
 
-KATANA_EXPORT std::string RandomAlphanumericString(uint64_t len);
-// Range 0..len-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t len);
-// Range min+1..max-1, inclusive
-KATANA_EXPORT int64_t RandomUniformInt(int64_t min, int64_t max);
-// Range 0.0f..max, inclusive
-KATANA_EXPORT float RandomUniformFloat(float max);
-
-KATANA_EXPORT std::mt19937& GetGenerator();
+using RandGenerator = std::mt19937;
+
+/// Generate a random alphanumeric string of length \param len using
+/// \param gen if provided. If no generator is specified, use the output of
+/// GetGenerator
+KATANA_EXPORT std::string RandomAlphanumericString(
+    uint64_t len, RandGenerator* gen = nullptr);
+
+KATANA_EXPORT RandGenerator& GetGenerator();
```

Comment:
Can you add a comment here that mentions that this returns a thread-local generator?

Reply:
Will do.